service/app/controllers/usermanagement/validator/UserGetRequestValidator.java

```diff
@@ -34,8 +31,13 @@ public void validateGetUserByKeyRequest(Request request) {
               JsonKey.KEY,
               key,
               String.join(StringFormatter.COMMA, JsonKey.EMAIL, JsonKey.PHONE, JsonKey.LOGIN_ID)));
+    }
 
+    if (request.get(JsonKey.KEY).equals(JsonKey.PHONE)) {
```

Comment:
what will happen if request.get(JsonKey.KEY) is null, revert the condition as follow.
if (JsonKey.PHONE.equals(request.get(JsonKey.KEY))) 
same applicable for phone as well.